src/ILLink.RoslynAnalyzer/XmlAttributeParser.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using System.Xml.XPath;
+using System.Reflection.Metadata;
+using System.Xml.Schema;
+using System.Collections.Generic;
+using System.Diagnostics;
+using ILLink.Shared;
+using System.Collections;
+using System.Net.NetworkInformation;
+using System.Data;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace ILLink.RoslynAnalyzer
```

Comment:
We should share this code between linker and analyzer. The XML format is the same, and that's all this code cares about. The specifics of converting the results into Cecil or Roslyn representations can be built on top of this.